presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomfilters.java

```diff
@@ -0,0 +1,332 @@
+import com.facebook.presto.orc.metadata.OrcMetadataReader;
+import com.facebook.presto.orc.metadata.RowGroupBloomfilter;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.predicate.Domain;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.primitives.Longs;
+import io.airlift.slice.Slice;
+import org.apache.hadoop.hive.ql.io.orc.OrcProto;
+import org.apache.hive.common.util.BloomFilter;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestOrcBloomfilters
+{
+    private static final String TEST_STRING = "ORC_STRING";
+    private static final String TEST_STRING_NOT_WRITTEN = "ORC_STRINGnot";
+    private static final String TEST_STRING_2 = "ORC_STRING_ALT";
+    private static final int TEST_INTEGER = 12345;
+
+    @Test
+    public void testHiveBloomfilterSerde()
+            throws Exception
+    {
+        BloomFilter bloomfilter = new BloomFilter(1_000_000L, 0.05D);
+
+        // String
+        bloomfilter.addString(TEST_STRING);
+        assertTrue(bloomfilter.testString(TEST_STRING));
+        assertFalse(bloomfilter.testString(TEST_STRING_NOT_WRITTEN));
+
+        // Integer
+        bloomfilter.addLong(TEST_INTEGER);
+        assertTrue(bloomfilter.testLong(TEST_INTEGER));
+        assertFalse(bloomfilter.testLong(TEST_INTEGER + 1));
+
+        // Re-construct
+        HiveBloomFilter hiveBloomFilter = new HiveBloomFilter(ImmutableList.copyOf(Longs.asList(bloomfilter.getBitSet())), bloomfilter.getBitSize(), bloomfilter.getNumHashFunctions());
+
+        // String
+        assertTrue(hiveBloomFilter.testString(TEST_STRING));
+        assertFalse(hiveBloomFilter.testString(TEST_STRING_NOT_WRITTEN));
+
+        // Integer
+        assertTrue(hiveBloomFilter.testLong(TEST_INTEGER));
+        assertFalse(hiveBloomFilter.testLong(TEST_INTEGER + 1));
+    }
+
+    @Test
+    public void testOrcHiveBloomfilterSerde()
```

Comment:
I totally agree, but the thing was it will generate quite a bit of duplicate code / lot of helper functions in the class. Basically the tests covers full serialization and deserialization in one sweep, which is in my opinion directly related. 

Actually it was smaller before but I had to expand it to cover a bug which I was unable to find otherwise. Please let me know what you'd prefer.